Sources/SwiftDocC/Model/DocumentationNode.swift

```diff
@@ -849,3 +852,45 @@ private extension BlockDirective {
         directivesSupportedInDocumentationComments.contains(name)
     }
 }
+
+extension [String] {
+
+    /// Strip the minimum leading whitespace from all the strings in this array, as follows:
+    /// - Find the line with least amount of leading whitespace. Ignore blank lines during this search.
+    /// - Remove that number of whitespace chars from all the lines (including blank lines).
+    /// - Returns: An array of strings with the minimum leading whitespace removed.
+    func linesWithoutLeadingWhitespace() -> [String] {
+
+        // Optimization: If the array of lines is empty then we return
+        guard !isEmpty else {
+            return self
+        }
+
+        // Optimization for the common case: If any of the lines do not start
+        // with whitespace, return the original lines.
+        for line in self {
+            if let firstChar = line.first, !firstChar.isWhitespace {
+                return self
+            }
+        }
```

Comment:
FYI: this can be written a bit more descriptively using `contains(where:)`

```suggestion
        self.contains(where: { $0.first?.isWhitespace == false })
```